src/pudl/transform/ferc1.py

```diff
@@ -5849,12 +5877,31 @@ def add_dimension_total_calculations(
 ) -> pd.DataFrame:
     """Define dimension total calculations.
 
-    In addition to calculations defining how values reported in one set of facts can
-    be aggregated resulting in a value in another fact, there are implied calculation
-    relationships between values reported within a dimension. In particular, when a
-    dimension reports a ``total`` value, we assume it is the sum of all the non-total
-    values for that dimension. This function makes these within-dimension calculation
-    relationships explicit by defining new calculations for each dimension's ``total``.
+    If a factoid is marked as a total along some dimension, then that's
+    implicitly a calculation. Make these explicit.
+
+    The multi-dimension totals logic is a little hard to follow, so a concrete
+    example:
+
+    If there is a record with "total" in utility_type and plant_status,
+    table_dimensions will say that that factoid has
+
+    * possible utility types: "total", "gas", "electric";
+    * possible plant status: "total", "in_service", "future"
+
+    In calc_components, we'll see these children for the "total", "total" parent.
+
+    * child:"factoid", "electric", "in_service"; parent: "factoid", "total", "total"
+    * child:"factoid", "electric", "future"; parent: "factoid", "total", "total"
+    * child:"factoid", "gas", "in_service"; parent: "factoid", "total", "total"
+    * child:"factoid", "gas", "future"; parent: "factoid", "total", "total"
+
+    We will also see these which are redundant with the above, and should be filtered out:
```

Comment:
Yes, this is all confusing. I think factoids that have `total` along any dimension should not be included in components - instead, we should be unrolling *those* into their own component parts. That actually seems like a pretty simple check to make, also - let me write that up.

Here are the situations where the parent and child both have one total dimension, and what I expect:

* different `total` dimensions:
  ```
  child: "factoid", "gas", "total"; parent: "factoid", "total", "leased"
  ```
  Should have no component relationship, because the `total` child status is bigger than the `leased` parent status. This holds whenever there's a `total` child dimension that is not also a `total` parent dimension.
* same `total` dimension:
  ```
  child: "factoid", "gas", "total"; parent: "factoid", "electric", "total"
  child: "factoid", "gas", "total"; parent: "factoid", "gas", "total"
  ```
  * The first one is not a match because the child utility type `gas` is not a conceptual subset of the parent utility type `electric`.
  * The second one is not a match because they're the same dimension

Reply:
That would be an error, also - I updated the docs / code to hopefully be more clear that we should *never* infer a calculation-component relationship where the child is a total in this function. 

We might still run into those in calculations that cross factoid boundaries, but that's not what we're dealing with here.